core/templates/components/code-mirror/codemirror-mergeview.component.ts

```diff
@@ -27,14 +28,24 @@ import { downgradeComponent } from '@angular/upgrade/static';
 export class CodemirrorMergeviewComponent implements
   AfterViewInit, OnInit, OnChanges {
   @Input() options = {};
-  @Input() leftValue;
-  @Input() rightValue;
-  codeMirrorInstance: CodeMirror.MergeView.MergeViewEditor;
-  constructor(private elementRef: ElementRef, private ngZone: NgZone) { }
+  // These properties comes from 'state-diff-modal.controller.ts' file and can
+  // be undefined sometimes, after the view initializes.
+  @Input() leftValue: string = ' ';
```

Comment:
Why do we initialize these? @srijanreddy98 I thought we discussed that `' '` is not a valid default value; these should ideally be initiailized with the correct values from the get-go.

Also the comment above this line doesn't seem to match the typing.

Reply:
These values comes from state-diff-modal as explained earlier, so no matter what we put here it will only be populated after the modal gets response for the request sent. I chose to make it string because earlier if they were undefined we assigned them to an empty `string`. If we leave this undefined, it will give error something like `properties are not initialised and definitely not assigned in the constructor` .So we have to again make it ` | undefined`, but we need to get rid of undefined here.Making it just undefined causes all the errors mentioned above. For this a ng-if check have also been added at the modal which checks for it to be not undefined i.e. the response received from http requests is not undefined.Does this makes sense ? Also the comment was irrelevant so removed it.

Reply:
@seanlip PTAL

Reply:
Yup don't make it undefined. But leave a comment in the code that clearly explains why you actually need to give it default values.

Is this a standard thing for all `@Input()`s -- i.e. they need default values? /cc @srijanreddy98

Reply:
@seanlip Not really a standard thing but angular also initialize it sometimes with empty string https://angular.io/guide/inputs-outputs#configuring-the-child-component we're using non-null assertion operator as used here https://angular.io/api/core/ViewChild#usage-notes like other places, this also works fine. We will have less ambiguous codebase this way!